ClemBot.Bot/bot/cogs/translate_cog.py

```diff
@@ -86,22 +24,39 @@ def __init__(self, bot):
 
     @ext.group(case_insensitive=True, invoke_without_command=True)
     @ext.long_help(
-        'Allows you to translate words or sentences by either specifying both the input and output language with the text to translate, or just the output language and the text to translate. run \'translate languages\' to see available languages')
+        'Allows you to translate words or sentences by specifying an output language and autodetecting the language you entered. Use !translate m for specifying the language you entered. Use \'translate languages\' to see available languages')
     @ext.short_help('Translates words or phrases between two languages')
-    @ext.example(('translate en spanish Hello', 'translate german Hello', 'translate languages', 'translate Spanish German Como estas?'))
+    @ext.example(('translate es Hello', 'translate german Hello', 'translate languages', 'translate German Como estas?, translate m german english I love partying with german people!'))
     async def translate(self, ctx, *input: str):
         if len(input) < 2:
             raise UserInputError("Incorrect Number of Arguments. Minimum of 2 arguments")
 
-        if is_valid_lang_code(input[1]):
-            await self.translate_given_lang(ctx, input)
+        if (is_valid_lang_code(input[0]) != None):
+          await self.translate_given_lang(ctx, input)
+        else:
+            raise UserInputError("Invalid input! Use !translate m for manual translation.")
+
+    @translate.command()
+    @ext.long_help('Alternative command to manually specify languages to convert to.')
+    @ext.short_help('!translate m <output lang> <input lang> <text>')
+    @ext.example(('!translate m english spanish Me encantan las fiestas con amigos'))
+
+    async def m(self, ctx, *input: str):
+        if len(input) < 3:
+            raise UserInputError("Incorrect Number of Arguments. Minimum number of 3 arguments")
+        if (is_valid_lang_code(input[0]) and is_valid_lang_code(input[1])) != None:
+            await self.alternative_translate_lang(ctx, input)
         else:
-            await self.translate_detect_lang(ctx, input)
+              raise UserInputError("Invalid input! Specify the language you entered.")
+
+        return
 
+          
     @translate.command()
     @ext.long_help('Shows all available languages to translate between')
     @ext.short_help('Shows available languages')
     @ext.example(('translate languages'))
+
```

Comment:
I don't think it'll break anything but I'm pretty sure everyone puts the decorators up against the function/method header instead of having a new line inbetween. (kinda looks weird but doesn't really matter)

Reply:
yes